spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -208,7 +243,15 @@ private RewriteManifests.Result rewriteManifests(ManifestContent content) {
     if (spec.isUnpartitioned()) {
       newManifests = writeUnpartitionedManifests(content, manifestEntryDF, targetNumManifests);
     } else {
-      newManifests = writePartitionedManifests(content, manifestEntryDF, targetNumManifests);
+      if (partitionSortColumns != null) {
```

Comment:
Should both of these branches be the same? Just one uses the default ordering and one doesn't? That is what we declared in the api above

Reply:
That makes sense. I was probably just hesitant to touch the main working path with a custom extractor, but I can make that switch.

Reply:
I think your change here is a great time to eliminate the other path as well